source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -570,25 +531,34 @@ ShaderGraphPtr ShaderGraph::createSurfaceShader(
             }
         }
 
-        GeomPropDefPtr geomprop = nodeDefInput->getDefaultGeomProp();
-        if (geomprop)
+        // Check if the input is a uniform
+        bool isUniform = nodeDefInput->getIsUniform();
+        if (isUniform)
         {
-            inputSocket->setGeomProp(geomprop->getName());
-            input->setGeomProp(geomprop->getName());
+            inputSocket->makeConnection(input);
         }
-
-        // If no explicit connection, connect to geometric node if a geomprop is used
-        // or otherwise to the graph interface.
-        const string& connection = nodeInput ? nodeInput->getOutputString() : EMPTY_STRING;
-        if (connection.empty())
+        else
         {
+            GeomPropDefPtr geomprop = nodeDefInput->getDefaultGeomProp();
```

Comment:
Geom props can only be connected if the input is not a uniform (preserving existing logic where Parameters) do not allow this. I assume we can revisit this if the geomprop is a uniform vs a varying input but leaving the logic as is for now.